todo_app/data/trello_items.py

```diff
@@ -0,0 +1,176 @@
+
+    response = requests.request("GET", url, headers=headers, params=query, timeout=10)
+    lists = {entry['name']: entry['id'] for entry in json.loads(response.text)}
+
+    # Now add the new item to the To Do list
+    # Can also check in the future if the item is Doing, and choose the appropriate list
+    url = "https://api.trello.com/1/cards"
+
+    headers = {"Accept": "application/json"}
+    query = {'idList': lists['To Do'], 'name': title,
+             'key': os.environ.get("TRELLO_KEY"), 'token': os.environ.get("TRELLO_TOKEN")}
+
+    response = requests.request("POST", url, headers=headers, params=query, timeout=10)
+    newCard = json.loads(response.text)
+
+    return {'id': newCard['id'], 'status': 'To Do', 'title': newCard['name']}
+
+def saveItem(item):
+    """
+    Updates an existing item in the session. If no existing item matches the ID of the specified item, nothing is saved.
+
+    Args:
+        item: The item to save.
+    """
+    url = f'https://api.trello.com/1/boards/{os.environ.get("TRELLO_BOARD")}/lists'
+    headers = {"Accept": "application/json"}
+    query = {'cards': 'open', 'key': os.environ.get("TRELLO_KEY"), 'token': os.environ.get("TRELLO_TOKEN")}
+
+    response = requests.request("GET", url, headers=headers, params=query, timeout=10)
+
+    lists = json.loads(response.text)
+    listIDs = {entry['name']: entry['id'] for entry in lists}
+    cards = []
+    for cList in lists:
+        for card in cList['cards']:
+            cards.append(card)
+    cardID = -1
+    for card in cards:
+        if card['idShort'] == item.id:
+            cardID = card['id']
+
+    if cardID == -1:
+        return item
+
+    url = f'https://api.trello.com/1/cards/{cardID}'
+    headers = {"Accept": "application/json"}
+    query = {'name': item.name, 'key': os.environ.get("TRELLO_KEY"), 'token': os.environ.get("TRELLO_TOKEN")}
+    query['idList'] = listIDs['Doing'] if item.status == 'Doing' else listIDs['To Do']
+
+    response = requests.request("PUT", url, headers=headers, params=query, timeout=10)
+    return item
+
+def deleteItem(itemID):
+    """
+    Deletes and existing item in the session. If no existing item matches the ID of the specified item, nothing is deleted.
+
+    Args:
+        id: The ID of the item to delete.
+    """
+    url = f'https://api.trello.com/1/boards/{os.environ.get("TRELLO_BOARD")}/lists'
+    headers = {"Accept": "application/json"}
+    query = {'cards': 'open', 'key': os.environ.get("TRELLO_KEY"), 'token': os.environ.get("TRELLO_TOKEN")}
+
+    response = requests.request("GET", url, headers=headers, params=query, timeout=10)
+
+    lists = json.loads(response.text)
+    cards = []
+    for cList in lists:
+        for card in cList['cards']:
+            cards.append(card)
+    cardID = -1
+    for card in cards:
+        if card['idShort'] == int(itemID):
+            cardID = card['id']
+
+    if cardID == -1:
+        return
```

Comment:
Similarly to the above, I think we could delete lines 118-135 which would:
* Not change the external behaviour
* Speed up the case in which we actually delete an item
* Have less code for us to read, test & maintain in future!

Reply:
And same as above, I can improve this by storing the long ID in the card item, saving an API call. Thanks for pointing this out.